From andrew@lunn.ch  Fri Dec 18 10:27:50 2009
From: Andrew Lunn <andrew@lunn.ch>
Date: Sat, 12 Dec 2009 23:39:43 +0100
Subject: Staging: batman-adv: stop persistent warnings if a device is deactivated.
To: gregkh@suse.de
Cc: devel@linuxdriverproject.org, Andrew Lunn <andrew@lunn.ch>, Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Message-ID: <1260657587-1135-3-git-send-email-andrew@lunn.ch>


Without this change we spam the kernel log on every packet received on
any other interface when an interface has been added, but is not yet
active, ie UP.

Signed-off-by: Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Signed-off-by: Andrew Lunn <andrew@lunn.ch>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/batman-adv/routing.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/drivers/staging/batman-adv/routing.c
+++ b/drivers/staging/batman-adv/routing.c
@@ -1177,12 +1177,8 @@ static int discard_one_packet(struct bat
 {
 	int result = -EAGAIN;
 
-	if (batman_if->if_active != IF_TO_BE_ACTIVATED) {
-		printk(KERN_WARNING
-		       "batman-adv:Could not read from deactivated interface %s!\n",
-		       batman_if->dev);
-	} else {
-		if (batman_if->raw_sock)
+	if ((batman_if->if_active == IF_TO_BE_ACTIVATED)
+		&& (batman_if->raw_sock)) {
 			result = receive_raw_packet(batman_if->raw_sock,
 						    packet_buff,
 						    PACKBUFF_SIZE);
